index.js

```diff
@@ -231,56 +190,101 @@ function setCumulusEnvironment(cumulusMessage, context) {
   safeSetEnv('ASYNCOPERATIONID', getAsyncOperationId(cumulusMessage));
 }
 
+/**
+ * Read the CMA output/results after sending a command
+ *
+ * @param {Object} readLine - configured readline object
+ * @param {*} errorObj - cma errorObject with stderr string buffer
+ * @returns {Promise<Object>} - Promise that resolves to a parsed JSON object
+ *                              from the CMA output
+ */
+async function getCmaOutput(readLine, errorObj) {
+  return new Promise((resolve, reject) => {
+    let buffer = '';
+    readLine.resume();
+    readLine.on('line', (input) => {
+      if (input.endsWith('<EOC>')) {
+        readLine.pause();
+        readLine.removeAllListeners('line');
+        const endInput = input.replace('<EOC>', '');
+        buffer += endInput;
+        resolve(JSON.parse(buffer));
+      }
+      else {
+        buffer += input;
+      }
+    });
+    readLine.on('close', () => {
+      reject(new CumulusMessageAdapterExecutionError(errorObj.stderr));
+    });
+  });
+}
+
 /**
  * Build a nested Cumulus event and pass it to a tasks's business function
  *
  * @param {Function} taskFunction - the function containing the business logic of the task
  * @param {Object} cumulusMessage - either a full Cumulus Message or a Cumulus Remote Message
  * @param {Object} context - an AWS Lambda context
- * @param {Function} callback - the callback to be called when the taskFunction
- *   has completed.  This should be the callback passed to the Lambda handler.
  * @param {string} schemas - Location of schema files, defaults to null.
- * @returns {undefined} - there is no return value from this function, but
- *   the callback function will be invoked with either an error or a full
- *   Cumulus message containing the result of the business logic function.
+ * @returns {Object} - The response from the call to createNextEvent or the taskFunction
+ *                     depending on the CUMULUS_MESSAGE_ADAPTER_DISABLED environment variable
  */
-function runCumulusTask(taskFunction, cumulusMessage, context, callback, schemas = null) {
-  let promisedNextEvent;
-  setCumulusEnvironment(cumulusMessage, context);
+async function runCumulusTask(taskFunction, cumulusMessage,
+  context, schemas = null) {
   if (process.env.CUMULUS_MESSAGE_ADAPTER_DISABLED === 'true') {
-    promisedNextEvent = invokePromisedTaskFunction(
+    const functionReturn = await invokePromisedTaskFunction(
       taskFunction,
       cumulusMessage,
       context
     );
+    return functionReturn;
   }
-  else {
-    const promisedRemoteEvent = loadAndUpdateRemoteEvent(cumulusMessage, context, schemas);
-    const promisedNestedEvent = promisedRemoteEvent.then((event) => {
-      // Reset the environment with what we can grab from the event from S3
-      setCumulusEnvironment(event, context);
-      return loadNestedEvent(event, context, schemas);
+  try {
+    const { cumulusMessageAdapter, errorObj } = await invokeCumulusMessageAdapter();
```

Comment:
@markdboyd  suggests calling it cmaProcess instead of cumulusMessageAdapter

Reply:
9a15062 -- I think this addresses yours and Marc's concerns.  Also re-enables lint in CI (why weren't we doing that?!)